src/fa/game_session.py

```diff
@@ -177,13 +178,41 @@ def _turn_state_changed(self, val):
 
     def _launched(self):
         self._logger.info("Game has started")
+        with open(util.LOG_FILE_GAME, 'r') as src, open(util.LOG_FILE_GAME+'.full.log', 'a+') as dst: dst.write(src.read())
 
     def _exited(self, status):
         self._game_connection = None
         self.state = GameSessionState.OFF
         self._logger.info("Game has exited with status code: {}".format(status))
         self.send('GameState', ['Ended'])
 
+        replayID=''
+        with open(util.LOG_FILE_GAME,'r') as inF:
+            for line in inF:
+                if 'gpgnet://localhost/' in line:
+                    try:
+                        replayID=line.split('localhost/')[1].split('/')[0]
+                    except:
+                        replayID='ERR'
+        try:
+            nb_replayid_files=0
+            list_game_replayid_files=os.listdir(util.LOG_DIR)
+            size_replayid_files=len(list_game_replayid_files)-1
+            while size_replayid_files >= 0:
+                if "Replayid" not in list_game_replayid_files[size_replayid_files]:
+                    list_game_replayid_files.remove(list_game_replayid_files[size_replayid_files])
+                else:
+                    nb_replayid_files+=1
+                size_replayid_files-=1
+            if nb_replayid_files >=11:
```

Comment:
This looks really arbitrary. Why 11 and why remove a single replay? If we want to rotate replay files, then let's do it where the main logfile is rotated. Also note that listdir returns entries in arbitrary order, so we'd have to make sure we order them by modification time or replay id before deleting the logs.

Reply:
Arbitrary indeed. To avoid having 200 log files for example, just keep the most recent ones, it can be 11 or 10 or something else (whatever number you think is fine).

And it removes only one because the limitation is called every time you end a game. (so at most you have +1 replay before the next time it is called)